pkg/api/validation/validation_test.go

```diff
@@ -1866,6 +1805,160 @@ func TestValidatePod(t *testing.T) {
 				DNSPolicy:     api.DNSClusterFirst,
 			},
 		},
+		{ // Serialized pod affinity in affinity requirements in annotations.
+			ObjectMeta: api.ObjectMeta{
+				Name:      "123",
+				Namespace: "ns",
+				// TODO: Uncomment and move this block into Annotations map once
+				// RequiredDuringSchedulingRequiredDuringExecution is implemented
+				//"requiredDuringSchedulingRequiredDuringExecution": [
+				//	{
+				//		"labelSelector":
+				//			{
+				//				"matchExpressions": [{
+				//					"key": "key2",
+				//					"operator": "In",
+				//					"values": ["value1", "value2"]
+				//				}]
+				//			},
+				//			"namespaces":[
+				//			{
+				//				"metadata":{
+				//					"name": "ns",
+				//					"labels":{
+				//						"key1": "value1",
+				//						"key2": "value2"
+				//					}
+				//				}
+				//			}],
+				//			"topologyKey": "zone"
+				//		}
+				//	]
+				Annotations: map[string]string{
+					api.AffinityAnnotationKey: `
+					{"podAffinity": {
+						"requiredDuringSchedulingIgnoredDuringExecution": [{
```

Comment:
I agree it's not very expressive, but I do believe it will be difficult for admins to craft sane constraints, which may be difficult to debug.  I'm trying to wear my admin + user hat before this gets into the wild.